src/structures/game_session.ts

```diff
@@ -259,6 +263,11 @@ export default class GameSession {
             this.endSession();
         } else if (this.scoreboard.gameFinished(guildPreference)) {
             this.endSession();
+        } else if (this.gameType === GameType.ELIMINATION) {
+            const eliminationScoreboard = this.scoreboard as EliminationScoreboard;
+            if (eliminationScoreboard.gameFinished()) {
```

Comment:
Isn't this the same as line 265

Reply:
Since the subclass and superclass have different signatures I think I have to do this. Should I add guildPreferences as an unused argument to elimination scoreboard and remove these lines?

Reply:
Let's do that

Reply:
Nevermind I'm wrong, I just tested it like you suggested and it works